pkg/controller/sriovnetwork/sriovnetwork_controller.go

```diff
@@ -141,33 +143,59 @@ func (r *ReconcileSriovNetwork) Reconcile(request reconcile.Request) (reconcile.
 	if err := controllerutil.SetControllerReference(instance, netAttDef, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
-
-	// Check if this NetworkAttachmentDefinition already exists
-	found := &netattdefv1.NetworkAttachmentDefinition{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: netAttDef.Namespace}, found)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
-			err = r.client.Create(context.TODO(), netAttDef)
-			if err != nil {
-				reqLogger.Error(err, "Couldn't create SriovNetworkNodeState", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
-				return reconcile.Result{}, err
+	if lastNetworkNamespace[instance.GetUID()] == "" {
+		lastNetworkNamespace[instance.GetUID()] = netAttDef.GetNamespace()
+	}
+	if lastNetworkNamespace[instance.GetUID()] == netAttDef.GetNamespace() {
+		// networkNamespace not updated
+		namespaces = netAttDef.GetNamespace()
+		// Check if this NetworkAttachmentDefinition already exists
+		found := &netattdefv1.NetworkAttachmentDefinition{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: namespaces}, found)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+				err = r.client.Create(context.TODO(), netAttDef)
+				if err != nil {
+					reqLogger.Error(err, "Couldn't create NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+					return reconcile.Result{}, err
+				}
+				lastNetworkNamespace[instance.GetUID()] = netAttDef.GetNamespace()
 			}
+			// NetworkAttachmentDefinition created successfully - don't requeue
+			return reconcile.Result{}, err
 		}
-		// NetworkAttachmentDefinition created successfully - don't requeue
-		return reconcile.Result{}, err
-	} else {
-		reqLogger.Info("SriovNetworkNodeState already exists, updating")
+		reqLogger.Info("NetworkAttachmentDefinition already exists, updating")
 		found.Spec = netAttDef.Spec
 		err = r.client.Update(context.TODO(), found)
 		if err != nil {
-			reqLogger.Error(err, "Couldn't update SriovNetworkNodeState", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+			reqLogger.Error(err, "Couldn't update NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
 			return reconcile.Result{}, err
 		}
+		// NetworkAttachmentDefinition updated successfully - don't requeue
+		return reconcile.Result{}, nil
 	}
-
-	// NetworkAttachmentDefinition already exists - don't requeue
-	reqLogger.Info("Skip reconcile: NetworkAttachmentDefinition already exists", "Namespace", found.Namespace, "Name", found.Name)
+	// networkNamespace updated, delete existing net-att-def cr and create a new one
```

Comment:
There seems some code overlap that:
no matter `lastNetworkNamespace[instance.GetUID()] == netAttDef.GetNamespace()` is satisfied or not, `r.client.Get` and `r.client.Create` will be called, can we combine the two piece of code and do the conditional check only for `r.client.Delete`?

Reply:
fixed